### ISSUE TITLE
Invalid import of a compiled enum from JS file into TS file

### ISSUE THREAD

**Issue by webschik** [alangpierce/sucrase#620]
Hello,

Thank you for developing this amazing project!

Recently I've tried to use `sucrase` in my project to speed up a build process and found a nasty bug.

If I try to import a compiled enum from a `.js` file (e.g. 3rd-party library) into my `.ts` file, the latter one gets an incorrect import.
Please check this example:

JS file:

```js
// enum1.js
"use strict";Object.defineProperty(exports, "__esModule", {value: true});var Enum1; (function (Enum1) {
    const KEY_1 = 'value1'; Enum1["KEY_1"] = KEY_1;
    const KEY_2 = 'value1'; Enum1["KEY_2"] = KEY_2;
})(Enum1 || (exports.Enum1 = Enum1 = {}));
```

This is a complied enum, see https://sucrase.io/#selectedTransforms=typescript,imports&code=export%20enum%20Enum1%20%7B%0A%20%20%20%20KEY_1%20%3D%20'value1'%2C%0A%20%20%20%20KEY_2%20%3D%20'value1'%0A%7D%0A

and the TS file:

```ts
// enum2.ts
import {Enum1} from './enum1.js';

enum Enum2 {
    KEY_1 = Enum1.KEY_1
}
```

If we compile `enum2.ts` with the next config 

```js
const {code} = transform(src, {
    transforms: ['typescript', 'imports'],
    filePath: './enum2.ts'
});
```

it will give us the next code

```js
 "use strict"; function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }// We need to import JS file
var _enum1js = require('./enum1.js'); var _enum1js2 = _interopRequireDefault(_enum1js);

var Enum2; (function (Enum2) {
    const KEY_1 = Enum1.KEY_1; Enum2[Enum2["KEY_1"] = KEY_1] = "KEY_1";
})(Enum2 || (Enum2 = {}));
```

As you may see `Enum2` depends on `Enum1` but the variable `Enum1` is not defined anywhere and not imported from `_enum1js`

**Comment by alangpierce** [comment]
Hi @webschik , thanks for reporting! It looks the issue is that the enum value expressions aren't being properly passed into the other Sucrase transforms (in this case the imports transform), which isn't a big deal when they're string/number literals but breaks down in cases like these. I'll take a closer look, hopefully should be a simple fix.

**Comment by alangpierce** [comment]
Just published a fix as 3.18.2

**Comment by webschik** [comment]
@alangpierce, thanks for the very quick fix!
It works! 🥳
